Apigee/ManagementAPI/Developer.php

```diff
@@ -369,11 +369,39 @@ public function setPageSize($size)
      */
     public function __construct(OrgConfig $config)
     {
-        $this->init($config, '/o/' . rawurlencode($config->orgName) . '/developers');
+        $this->organizationName = $config->orgName;
+        $this->init($config, '/o/' . rawurlencode($this->organizationName) . '/developers');
         $this->blankValues();
         $this->pageSize = self::MAX_ITEMS_PER_PAGE;
     }
 
+    /**
+     * Gets the static cache.
+     *
+     * @param string $organization
+     *   The requested organization.
+     * @param bool $reset
+     *   Resets the static cache for the requested organization.
+     *
+     * @return array
+     *   Returns the static cache for the requested organization.
+     */
+    protected static function &getStaticCache($organization, $reset = false)
+    {
+        static $data = array();
+
+        // Initialize static caches.
+        if (!array_key_exists($organization, $data) || $reset) {
+            $data[$organization] = array(
+                'developers' => array(),
+                'all_cached' => false,
+                'all_expanded' => false,
```

Comment:
Why would you cache non-expanded developer objects? That is just an other complexity that you have to deal with. Developer cache should contain only fully loaded developer objects.

Also, if you would like to use a complex structure like this for static cache I'd rather recommend you to create an object for that and instead of this array. It is not just complicated to work with an array, but also can cause performance degradation.

https://steemit.com/php/@crell/php-use-associative-arrays-basically-never

Reply:
Also, are you sure that we should introduce the static cache in the PHP SDK level not just in the Drupal 7 module? I do not know how many 3rd party apps relies on the V1 SDK but maybe we could handle this performance problem only in the Drupal 7 module. Therefore we would only need to deal with possible problems (bugs) that these changes may introduce in one place instead of two. 
(Just an idea.)
PS.: I haven't implemented static caching in the V2 PHP API client, because if consumer apps needs that they can easily add that and after that they have to deal with the cache invalidation problem. ;)